.yarn/patches/@metamask-providers-npm-18.2.0-3616fe3170.patch

```diff
@@ -0,0 +1,188 @@
+                 method,
+                 params,
+             };
+diff --git a/dist/BaseProvider.mjs b/dist/BaseProvider.mjs
+index 2a88b945be89d73d04a075508d6f2858a8c45feb..e371b418891d07e644866b521ad552425a25182a 100644
+--- a/dist/BaseProvider.mjs
++++ b/dist/BaseProvider.mjs
+@@ -20,6 +20,7 @@ import { JsonRpcEngine } from "@metamask/json-rpc-engine";
+ import { rpcErrors, JsonRpcError } from "@metamask/rpc-errors";
+ import SafeEventEmitter from "@metamask/safe-event-emitter";
+ import $dequal from "fast-deep-equal";
++import { v4 as uuidV4 } from "uuid";
+ const dequal = $importDefault($dequal);
+ import messages from "./messages.mjs";
+ import { getRpcPromiseCallback, isValidChainId } from "./utils.mjs";
+@@ -135,9 +136,11 @@ export class BaseProvider extends SafeEventEmitter {
+         }
+         const payload = params === undefined || params === null
+             ? {
++                id: uuidV4(),
+                 method,
+             }
+             : {
++                id: uuidV4(),
+                 method,
+                 params,
+             };
+diff --git a/dist/StreamProvider.cjs b/dist/StreamProvider.cjs
+index 340e5179e02bedc6854b0d7b5a8861bab101b5ec..a1b9d21395c8027c12778933337aaaa55c6717f9 100644
+--- a/dist/StreamProvider.cjs
++++ b/dist/StreamProvider.cjs
+@@ -29,23 +29,27 @@ class AbstractStreamProvider extends BaseProvider_1.BaseProvider {
+      * listeners. Default: 100.
+      * @param options.rpcMiddleware - The RPC middleware stack to use.
+      */
+-    constructor(connectionStream, { jsonRpcStreamName, logger = console, maxEventListeners = 100, rpcMiddleware = [], }) {
++    constructor(connectionStream, { jsonRpcStreamName = 'metamask-provider', logger = console, maxEventListeners = 100, rpcMiddleware = [], multiplexing = true, }) {
+         super({ logger, maxEventListeners, rpcMiddleware });
+         if (!(0, is_stream_1.duplex)(connectionStream)) {
+             throw new Error(messages_1.default.errors.invalidDuplexStream());
+         }
+         // Bind functions to prevent consumers from making unbound calls
+         this._handleStreamDisconnect = this._handleStreamDisconnect.bind(this);
+-        // Set up connectionStream multiplexing
+-        const mux = new object_multiplex_1.default();
+-        (0, readable_stream_1.pipeline)(connectionStream, mux, connectionStream, this._handleStreamDisconnect.bind(this, 'MetaMask'));
+         // Set up RPC connection
+         // Typecast: The type of `Duplex` is incompatible with the type of
+         // `JsonRpcConnection`.
+         this._jsonRpcConnection = (0, json_rpc_middleware_stream_1.createStreamMiddleware)({
+             retryOnMessage: 'METAMASK_EXTENSION_CONNECT_CAN_RETRY',
+         });
+-        (0, readable_stream_1.pipeline)(this._jsonRpcConnection.stream, mux.createStream(jsonRpcStreamName), this._jsonRpcConnection.stream, this._handleStreamDisconnect.bind(this, 'MetaMask RpcProvider'));
++        if(multiplexing) {
```

Comment:
The multiplexing option was introduced because the UI and background are already creating a multiplex (see [this UI code](https://github.com/MetaMask/metamask-extension/pull/29650/files#diff-5a9d964db9d7c2fe207e21470884b2710e54723e0c1e6e6f7fd264706604fd6cR340)). This new parameter in StreamProvider enables it to use the existing stream directly, instead of creating an additional multiplex.